src/adapter/src/catalog/storage.rs

```diff
@@ -1898,14 +1955,16 @@ pub struct ItemValue {
     definition: SerializedCatalogItem,
 }
 
-#[derive(Clone, Deserialize, Serialize, PartialOrd, PartialEq, Eq, Ord, Hash)]
+#[derive(Clone, Deserialize, Serialize, PartialOrd, PartialEq, Eq, Ord, Hash, Debug)]
 pub struct RoleKey {
     id: RoleId,
 }
 
-#[derive(Clone, Deserialize, Serialize, PartialOrd, PartialEq, Eq, Ord)]
+#[derive(Clone, Deserialize, Serialize, PartialOrd, PartialEq, Eq, Ord, Debug)]
 pub struct RoleValue {
-    name: String,
+    // flatten needed for backwards compatibility.
+    #[serde(flatten)]
+    role: SerializedRole,
```

Comment:
This struct may be pointless now, and we should instead just inline it's contents here.